src/Modules/OrchardCore.Commerce.Promotion/Views/DiscountPart.cshtml

```diff
@@ -0,0 +1,62 @@
+@using OrchardCore.Commerce.ContentFields.ViewModels
+@using OrchardCore.Commerce.MoneyDataType
+@using OrchardCore.ContentManagement.Metadata.Models
+
+@model OrchardCore.Commerce.Promotion.ViewModels.DiscountPartViewModel
+
+@{
+    var discountPart = Model.DiscountPart;
+}
+
+@if (discountPart.IsValidAndActive())
+{
+    @if (discountPart.DiscountPercentage is { } discountPercentage && discountPercentage.Value > 0)
```

Comment:
All of the `DiscountPart` properties are initialized with `new()` so you don't have to check if they are `null` in this file.

Reply:
I will keep the `null` check for the DateField values, since the field can't be `null`, but it's value property can.
![image](https://user-images.githubusercontent.com/80963259/204001729-8c9c0e6a-4848-44bd-8d30-53441512ef40.png)